pkg/profilecreator/profilecreator.go

```diff
@@ -344,6 +344,34 @@ func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, sp
 	return ghwHandler.getCPUsSequentially(reservedCPUCount, htEnabled, topologyInfo.Nodes)
 }
 
+type cpuAccumulator struct {
+	builder *cpuset.Builder
+	count   int
+}
+
+func newCPUAccumulator() *cpuAccumulator {
+	return &cpuAccumulator{
+		builder: cpuset.NewBuilder(),
+	}
+}
+
+// AddCores adds logical cores from the slice of *cpu.ProcessorCore to a CPUset till the cpuset size is equal to the max value specified
+// In case the max is specified as allCores, all the cores from the slice of *cpu.ProcessorCore are added to the CPUSet
+func (ca *cpuAccumulator) AddCores(max int, cores []*cpu.ProcessorCore) {
```

Comment:
We might want to add a unit test for AddCores. your call

Reply:
done